Put webRequestTranslator is webRDFS folder

vnazz opened the pull request:
WebRequestTranslator no longer has its own folder.

vnazz committed:
Put webRequestTranslator is webRDFS folder

vnazz committed:
Put webRequestTranslator is webRDFS folder

LINKIWI reviewed (approved):
Nice, thanks, lgtm if compile is green

While you're here, can you rename the files to follow the existing convention:

```
webRequestTranslator.h -> web_request_translator.h
webRequestTranslator.cc -> web_request_translator.cc
```

vnazz committed:
Rename translator to match style

vnazz committed:
Rename translator to match style

jessicayu626 committed:
Feature/leases (#150)

* Add native file systems tests.

* pull

* Implement renewlease, recoverlease and complete.

* fix client path errors.

* Add WebRDFS HTTP server library to build (#141)

* Add Python-based linter virtualenv to .gitignore

* Simple-Web-Server in utility

* Rename library hpp -> h, cpp -> cc and remove examples from CMakeLists.txt

* Feature/webrdfs web request translator (#148)

* Add basic structure for webRequestTranslator and tests

* Add read, status and listing to webRequestTranslator

* add http responses for create, mkdir, mv delete

* Fix build issue on travis

* Hopefully fix permission issue in translator

* Jk this should fix permissions

* Add newlines to fix lint

* Add WebRDFS HTTP server library to build (#141)

* Add Python-based linter virtualenv to .gitignore

* Simple-Web-Server in utility

* Rename library hpp -> h, cpp -> cc and remove examples from CMakeLists.txt

* Change directory name and add more info to mv, delete, mkdir handlers

* Add basic structure for webRequestTranslator and tests

* Add read, status and listing to webRequestTranslator

* add http responses for create, mkdir, mv delete

* Fix build issue on travis

* Hopefully fix permission issue in translator

* Jk this should fix permissions

* Add newlines to fix lint

* Change directory name and add more info to mv, delete, mkdir handlers

* Replace underscores

* Update Zookeeper mirror to Rice server (#167)

* Temporarily disable RW test (#168)

* WebRDFS server process, placeholder HTTP handlers (#144)

* Simple-Web-Server in utility

* Add web-rdfs directory

* Main executable, handler registration

* Address lint errors

* Sample handlers registered

* send proto to zk in webrdfs

* Rename library hpp -> h, cpp -> cc and remove examples from CMakeLists.txt

* Correctly reference header files

* Lint errors

* add leasetest.

* Add lease tests.

* Add WebRDFS HTTP server library to build (#141)

* Add Python-based linter virtualenv to .gitignore

* Simple-Web-Server in utility

* Rename library hpp -> h, cpp -> cc and remove examples from CMakeLists.txt

* Feature/webrdfs web request translator (#148)

* Add basic structure for webRequestTranslator and tests

* Add read, status and listing to webRequestTranslator

* add http responses for create, mkdir, mv delete

* Fix build issue on travis

* Hopefully fix permission issue in translator

* Jk this should fix permissions

* Add newlines to fix lint

* Add WebRDFS HTTP server library to build (#141)

* Add Python-based linter virtualenv to .gitignore

* Simple-Web-Server in utility

* Rename library hpp -> h, cpp -> cc and remove examples from CMakeLists.txt

* Change directory name and add more info to mv, delete, mkdir handlers

* Add basic structure for webRequestTranslator and tests

* Add read, status and listing to webRequestTranslator

* add http responses for create, mkdir, mv delete

* Fix build issue on travis

* Hopefully fix permission issue in translator

* Jk this should fix permissions

* Add newlines to fix lint

* Change directory name and add more info to mv, delete, mkdir handlers

* Replace underscores

* Fix lint errors.

* Fix build error

* Fix more lint errors.

* Two more lint errors.

* WebRDFS server process, placeholder HTTP handlers (#144)

* Simple-Web-Server in utility

* Add web-rdfs directory

* Main executable, handler registration

* Address lint errors

* Sample handlers registered

* send proto to zk in webrdfs

* Rename library hpp -> h, cpp -> cc and remove examples from CMakeLists.txt

* Correctly reference header files

* Lint errors

* Fix lease test.

* Sprint 1: Permissions on a file (#92)

* Add getUserName method in rpcServer and simple test

* added user permissions listing to FileZNode and initialized permissions
fields during file creation in ZK.

* Set owner HDFS call modifies permissions ACL in ZK; fixes #53

* Modify zk_nn to pass along client's username as identity to zk. Add bool indicating whether in secure mode in zk_nn. Check access before serving files.

* debug

* Fix username test so passes both locally and on travis and renamed function getUserName to getUsername

* added user permissions listing to FileZNode and initialized permissions
fields during file creation in ZK.

* Set owner HDFS call modifies permissions ACL in ZK; fixes #53

* Fix merge issue on .ci/test/main.sh

* adding zookeeper metadata for keeping admin identity (#56)
adding logic to process change of file permissions (#55)

* Fix typo in name node main

* Fix segfault

* Statically-sized char arrays in ACL

* Fix lint errors

* Fix lint issues

* Order of header lint error fixed

* Fix username test lint issues

* All remaining lint errors addressed

* merging permissions changes

* lint errors after merge

* add docs for zk_nn_client.h

* fix on logic to determin admin

* fix lint

* Add getUserName method in rpcServer and simple test

* added user permissions listing to FileZNode and initialized permissions
fields during file creation in ZK.

* added user permissions listing to FileZNode and initialized permissions
fields during file creation in ZK.

* Set owner HDFS call modifies permissions ACL in ZK; fixes #53

* Set owner HDFS call modifies permissions ACL in ZK; fixes #53

* Modify zk_nn to pass along client's username as identity to zk. Add bool indicating whether in secure mode in zk_nn. Check access before serving files.

* debug

* Fix username test so passes both locally and on travis and renamed function getUserName to getUsername

* Fix merge issue on .ci/test/main.sh

* adding zookeeper metadata for keeping admin identity (#56)
adding logic to process change of file permissions (#55)

* Fix typo in name node main

* Fix segfault

* Statically-sized char arrays in ACL

* Fix lint errors

* Add getUserName method in rpcServer and simple test

* Fix lint issues

* Order of header lint error fixed

* Fix username test lint issues

* All remaining lint errors addressed

* merging permissions changes

* lint errors after merge

* add docs for zk_nn_client.h

* fix on logic to determin admin

* fix lint

* fixing error made while fixing rebase conflicts

* generate cert (#171)

(cherry picked from commit 5109faaad3e7f61d057c4292ce5360fb4e089bef)

* Fix commit error.

* Fix build errors.

* Fix build error

* Debug test.

* Fix lint errors.

* Fix lint error.

* Add lease test executable.

LINKIWI committed:
WebRDFS HTTPS/SSL support (#176)

* Remove keys

* Link ssl and crypto in compile

* Convert HTTPServer -> HTTPSServer

* Absolute path to server certificate and key

* Add keys

* Fix compile error: no matching WebRDFSServer constructor

* Fix lint error

heliumj committed:
Fix typo (#181)

willkoh committed:
Added changes to vagrant_provision.sh to ensure reprovisioning works. (#182)

* Added changes to vagrant_provision.sh to ensure reprovisioning works.

* Added spacing to ensure bash is happy.

vnazz committed:
Feature/webrdfs delete read (#174)

* Got delete handler to work

* Add read functionality

* Remove unneeded zk in server

* Add delete test

* Add read test

* Change ci to use WebServerTest

* Fix lint issues

* Fix WebServerTest lint error

* Use actual webHDFS api and correct response

* Got delete handler to work

* Add read functionality

* Remove unneeded zk in server

* Add delete test

* Add read test

* Change ci to use WebServerTest

* Fix lint issues

* Fix WebServerTest lint error

* Use actual webHDFS api and correct response

* Fixed accidental http change

* Fix tests to use https

* Fix lint in test

* Change test to use existing files

* Fix delete test issue

cannontwo committed:
Bumped max number of client connections in ZooKeeper config (#183)

* Bumped max number of client connections in ZooKeeper config

* Added configuration line to Travis config as well

Commit:
Caching for getListing (#147)

* initialized lru cache

* forgot to add the lrucache header file

* surface cache implementation

* added caching test and some extra methods to zk nn client to support testing

* added another test to CacheTest

* cache ptr error

* fixed some compilation errors

* added more logging

* compile fix; still segfaults

* getting a double free error now

* pointer fix

* test fix + cache remove fix

* fixed seg fault that was happening on more than one cache check

* changed cmake faile

* resetting the logging flag to false

* cache methods got removed from zk_nn_client header somehow

* commented out one test

* stopping inifinite loop

* Teardown delete files

* close zk on teardown

* frivolous / faulty error -> info

* lint errors

Commit:
caching for ZkWrapper->get (#180)

* added initial cache stuff

* lru cache file

* fix indentation

* lru cache cmake

* added cache code to get

* finished cache code for zk wrapper get

* cache watch

* fixed include errors

* fixed linking errors

* fixed some tests and added more logging

* more testing logs

* close zk on teardown

* fixed all tests

* lint errors and rebase

Commit:
Feature/ec (#186)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

blockIndices optional field added for EC purpose.

a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

new proto messages related to a new set of EC RPC calls. (#122)

Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

Hns test fix (#139)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

* blockIndices optional field added for EC purpose.

* a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

* remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

* Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

* new proto messages related to a new set of EC RPC calls. (#122)

* Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

* a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

* hopefully fixed branhc

* can't find get_num_data_parity_blocks, everything else builds

* fixed lint error

* test passes

* fixed lint error

* another lint error

added isEC field to the FileZNode. (#149)

Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

blockIndices optional field added for EC purpose.

a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

new proto messages related to a new set of EC RPC calls. (#122)

Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

Hns test fix (#139)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

* blockIndices optional field added for EC purpose.

* a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

* remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

* Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

* new proto messages related to a new set of EC RPC calls. (#122)

* Fixed error with vagrant provisioning! (#137)

* Fixed error with vagrant provisioning!

* Fixed an error in travis building when downloading the ISAL.

* a new set of NameNode RPC APIs (#123)

* an namenode api that returns the set of registered EC policies implemented.

* register getErasureCodingPolicies in the translator

* hopefully fixed branhc

* can't find get_num_data_parity_blocks, everything else builds

* fixed lint error

* test passes

* fixed lint error

* another lint error

added isEC field to the FileZNode. (#149)

Feature/EC_Worker (#154)

* Feature/protobuff changes for ec (#59)

* cmake comment out the test part

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* added configurable node selection protocol (#79)

* added configurable node selection protocol

* added tiebreakers to TargetDN < operator

* added @param documentation for extra command line option

* fixed lint errors pt. 1

* fixed error message

* fix the build erorr.

* proto changes for EC based file creation.

* Get rid of changes to the test/CMakeLists.txt

* File creation related code changes. more changes to follow..

* Typo removed.

* fix the build erorr.

* lint fix.

* header lint fix

* lint fix2.

* blockIndices optional field added for EC purpose.

* a fix for ecPolicyName. also implemented the hierarchical naming func… (#90)

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* some starnge build error fix

* weird bug fix.

* blockIndices optional field added for EC purpose.

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* a fix for ecPolicyName. also implemented the hierarchical naming function

* bug fix. set the highest bit of a block group id to be 1

* set the highest bit of non EC storage blocks to 0. remove wrong comments

* add block group helper method put in place. must be further implemented.

* rename util:generate_uuid to generate_block_id

* create api now sets the default EC policy.

* some lint fix.

* remove all the hard coded assumptions that block metadata locations (#113)

* remove all the hard coded assumptions that block metadata are located at /block_locations/block_id.

* build fix.

* fix build error

* for some reason storing std::string in FileZNode causes errors runtime. comment them out for now.

* more refactoring on the datanode side code.

* Feature/EC-ISAL adds ISA-L into RDFS (#121)

* Addition of ISA-L wrapper.

* Modified CMake to build ISAL wrapper correctly.

* Added ErasureCodeTest files.

* Added scripts to pass travis build checks.

* Added tests. Also added scripts for travis build checks.

* Modified vagrant_provision to make sure vagrant is set up correctly.

* Deleted unnecessary commented line in tests.

* Changed links for isal.tar.gz to the one on Rice Server.

* chmod +x isal.sh

* Fixed minor spelling error present.

* Added making of the ISAL library into .ci/install/isal.sh

* new proto messages related to a new set of EC RPC calls. (#122)

* Added constants to zk_client_common for new ec_recover workqueue.

* Fixed typos.

* very in progress

* Added constants to zk_client_common for new ec_recover workqueue.

* Fixed typos.

* very in progress

* Watches health of datanodes and added functionality to post ec_recover
jobs.

* Fixed error of not casting block_id received from zk to uint64_t

* filled out handleReconstructCmds

* minor fixes

* started daemon thread in dn main.cc

* Added check_acks functionality for EC

* Fixed build errors.

* made remote reads modular

* fixed build errors

* reconstruction reads building

* added writeBlock method

* Added recovery computations.

* Made lint changes

* Added constants to zk_client_common for new ec_recover workqueue.

* Fixed typos.

* Added constants to zk_client_common for new ec_recover workqueue.

* very in progress

* Watches health of datanodes and added functionality to post ec_recover
jobs.

* Fixed error of not casting block_id received from zk to uint64_t

* filled out handleReconstructCmds

* minor fixes

* started daemon thread in dn main.cc

* Added check_acks functionality for EC

* Fixed build errors.

* made remote reads modular

* fixed build errors

* reconstruction reads building

* added writeBlock method

* Added recovery computations.

* Made lint changes

* Fixing lint.

added setPolicy proto (#166)

fixed block release

add block completed. (#151)

* add block completed.

* fixes the block generation id test.

* changes made based on comments.

* unit test added and bugs fixed.

* add block completed.

* fixes the block generation id test.

* changes made based on comments.

* unit test added and bugs fixed.

* add one more test and changes that populate the blockindices field.

* building, but tests fail

* fixed merge error

* fixed lint errors

add get ec policy of a path api and unit tests (#177)

* add get ec policy of a path api and unit tests

* set erasurecoindg policy implemented.

* set a src in the unit test.

added ec logic to blockReceived (#184)

* merging

* lint fixes

vnazz committed:
Rebase

vnazz committed:
Fix lint errors
